src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/_params.py

```diff
@@ -46,6 +46,20 @@
 register_cli_argument('resource feature list', 'resource_provider_namespace', CliArgumentType(options_list=('--namespace',), required=False, help=_PROVIDER_HELP_TEXT))
 register_cli_argument('resource feature', 'feature_name', CliArgumentType(options_list=('--name', '-n'), help='the feature name'))
 
+register_cli_argument('resource policy', 'resource_group_name', help='the resource group where the policy will be applied')
```

Comment:
>'resource_group_name', help='the resource group where the policy will be applied') [](start = 41, length = 82)

You do still need the 'arg_type=resource_group_name_type', you just don't need the completer part. Right now you've essentially defined a type with just help text.